components/ResourceIDs.tsx

```diff
@@ -0,0 +1,56 @@
+import { Button } from "@mantine/core";
+import Link from "next/link";
+import { ResourceTypeResponse, EntryKeyObject } from "./../pages/[resourceType]";
+
+/**
+ * Component for displaying resource IDs as Links
+ * @param props include jsonBody which is the response from a GET request to
+ * a resourceType endpoint.
+ * @returns an array of Links of resource IDs, or if none exist, a "No resource found" message, or if an
+ * invalid response body is passed through, an error message
+ */
+function ResourceIDs(props: { jsonBody: ResourceTypeResponse }) {
+  const entryArray = props.jsonBody.entry;
+
+  if (props.jsonBody.total === 0) {
+    return <div>No resources found</div>;
+  } else if (props.jsonBody.total > 0) {
+    return (
+      <div>
+        <h2>Resource IDs:</h2> {getAllIDs(entryArray)}
+      </div>
+    );
+  } else {
+    return <div>Error</div>;
+  }
+}
+
+/**
+ * Maps an array of resource bodies to an array of Link-wrapped-Buttons, where each Button represents a resource ID
+ * @param entry which is the entry key from a resource bundle. It is an array where each index is a resource's body
+ * @returns an array of Resource ID Link-wrapped-Buttons
+ */
+const getAllIDs = (entry: EntryKeyObject[]) => {
+  return entry.map((el) => {
+    return (
+      <Link href={`/`} key={el.resource.id} passHref>
+        <div>
+          <Button
+            color="cyan"
+            radius="md"
+            size="md"
+            variant="subtle"
+            style={{
+              padding: "2px",
+            }}
+          >
+            <div> {el.resource.id} </div>
```

Comment:
I think it'd be good to preface this ID with the resource type as well, since it's easy to forget which one you clicked on. Typical FHIR-like way to do this is with the string `<resourceType>/<id>`, so you can add a `{el.resource.resourceType}/` before rendering the ID, so it shows up as `Encounter/encounter-id`, `Procedure/procedure-id`, etc.